packages/vm/examples/compile_module.rs

```diff
@@ -0,0 +1,32 @@
+use std::fs::File;
+use std::io::Read;
+
+use clap::{App, Arg};
+
+use cosmwasm_vm::internals::compile;
+
+pub fn main() {
+    let matches = App::new("Module compilation")
+        .version("0.1.0")
+        .long_about("Checks that the given wasm file compiles successfully")
+        .author("Mauro Lacy <mauro@lacy.com.es>")
+        .arg(
+            Arg::with_name("WASM")
+                .help("Wasm file to read and compile")
+                .required(true)
+                .index(1),
+        )
+        .get_matches();
+
+    // File
+    let path = matches.value_of("WASM").expect("Error parsing file name");
+    let mut file = File::open(path).unwrap();
+
+    // Read wasm
+    let mut wasm = Vec::<u8>::new();
+    file.read_to_end(&mut wasm).unwrap();
+
+    // Compile module
+    compile(&wasm, None).unwrap();
```

Comment:
This will include the deterministic middleware, correct?
But not the imports checks?

It would be nice to do that as well, ensure it is compatible with this versions of cosmwasm plus. There is a function used somewhere when loading a file in cache

Reply:
> This will include the deterministic middleware, correct?

Correct.

> But not the imports checks?

Not sure about this.

> It would be nice to do that as well, ensure it is compatible with this versions of cosmwasm plus. There is a function used somewhere when loading a file in cache

Good point, I'll take a look at the compiling / execution sequence, and add the checks found there.

Reply:
For import checks, I would have to do something similar to `make_instance`: https://github.com/CosmWasm/cosmwasm/blob/ea7603d717c3d1c07046829e8ed3fb9c5c596599/packages/vm/src/environment.rs#L418.
There's also a similar one in `imports.rs`: https://github.com/CosmWasm/cosmwasm/blob/ea7603d717c3d1c07046829e8ed3fb9c5c596599/packages/vm/src/imports.rs#L557

These are test functions, and I would have to basically rewrite it / duplicate it for the utility. Wonder if there's a way to publish a test function, and use it somewhere else. That, or transforming this in a utility function somewhere, and use it from all places. This is good because it's less code maintenance.

Reply:
No, this is not needed. There's a `check_wasm` function that can be used.